Development oneapi branch for first ONEAPI release !!

shahzadATintel opened the pull request:
Development oneapi branch for first ONEAPI release !!

shahzadATintel committed:
created development_oneapi merge for oneapi first release

shahzadATintel committed:
tuned emulation switch to off for defaultwq

shahzadATintel committed:
updated yml file for CI

shahzadATintel committed:
updated yml file

shahzadATintel committed:
update version to 2.0-rc1

ymeng-git committed:
cleanup to prepare for 2.0-rc1 release

ymeng-git committed:
for oneAPI, use dpcpp not g++ compiler

ymeng-git committed:
Fixed formatting

ymeng-git committed:
disabled slow emulation run on seal test

shahzadATintel committed:
update ci for examples

shahzadATintel committed:
mering example ci updates
Merge branch 'development_oneapi' of https://github.com/intel/hexl-fpga into development_oneapi

shahzadATintel committed:
ran clang format

ymeng-git committed:
update ci for examples

ymeng-git committed:
Merge branch 'development_oneapi' of https://github.com/intel/hexl-fpga into development_oneapi

ymeng-git committed:
fixed formatting

shahzadATintel committed:
added docy doc for dl_kernel_interface class

ymeng-git committed:
added dpcpp as compiler

shahzadATintel committed:
Merge branch 'development_oneapi' of https://github.com/intel/hexl-fpga into development_oneapi

shahzadATintel committed:
addes description for enabling emulation

ymeng-git committed:
user to specify -DCMAKE_CXX_COMPILER=dpcpp in compilation

ymeng-git committed:
Merge branch 'development_oneapi' of https://github.com/intel/hexl-fpga into development_oneapi

ymeng-git committed:
bridge test to use the right bsp

ymeng-git commented on device/CMakeLists.txt:
```diff
@@ -1,25 +1,210 @@
+        target_link_options(${KERNEL} PRIVATE ${EMU_LINK_FLAGS})
+    endfunction()
+
+    build_standalone_emu_shared_lib(dyadic_multiply)
```
What are the final dpcpp compilation and linking commands we want to achieve for emulation/hw/report?

wangyon1 committed:
add link_options for keyswitch and compile options to avoid warning

wangyon1 committed:
fix whitespace error

wangyon1 committed:
change dyadic_multiply_keyswitch build target USM BSP

ymeng-git committed:
recreated/simplified the device build flow
added Steve into the contributor list
enabled dyadic_multiply_keyswitch test
updated CI
updated documents

ymeng-git committed:
fixed example run in CI

ymeng-git commented:
Closed this branch as the commits have been signed and squashed into the dev/oneapi branch, which has been merged into development.
